app/presenters/register_api/single_application_presenter.rb

```diff
@@ -0,0 +1,247 @@
+module RegisterAPI
+  class SingleApplicationPresenter
+    UCAS_FEE_PAYER_CODES = {
+      'SLC,SAAS,NIBd,EU,Chl,IoM' => '02',
+      'Not Known' => '99',
+    }.freeze
+
+    def initialize(application_choice)
+      @application_choice = ApplicationChoiceExportDecorator.new(application_choice)
+      @application_form = application_choice.application_form
+    end
+
+    def as_json
+      {
+        id: application_choice.id.to_s,
+        type: 'application',
+        attributes: {
+          support_reference: application_form.support_reference,
+          status: status, # keep to show offer withdrawn and deferred
+          updated_at: application_choice.updated_at.iso8601,
+          submitted_at: application_form.submitted_at.iso8601,
+          recruited_at: application_choice.recruited_at.iso8601,
+          candidate: {
+            id: "C#{application_form.candidate.id}",
+            first_name: application_form.first_name,
+            last_name: application_form.last_name,
+            date_of_birth: application_form.date_of_birth,
+            nationality: application_choice.nationalities,
+            domicile: application_form.domicile,
+            uk_residency_status: uk_residency_status,
+            uk_residency_status_code: uk_residency_status_code,
+            fee_payer: provisional_fee_payer_status,
+            english_main_language: application_form.english_main_language,
+            english_language_qualifications: application_form.english_language_qualification_details,
+            other_languages: application_form.other_language_details,
+            disability_disclosure: application_form.disability_disclosure, # do we need this?
+            gender: equality_and_diversity_data['sex'],
+            disabilities: equality_and_diversity_data['disabilities'].presence || [],
+            ethnic_group: equality_and_diversity_data['ethnic_group'],
+            ethnic_background: equality_and_diversity_data['ethnic_background'],
+          },
+          contact_details: contact_details,
+          course: course_info_for(application_choice.offered_option),
+          provider: provider_info_for(application_choice.offered_option.course.provider),
+          qualifications: qualifications,
+          hesa_itt_data: hesa_itt_data.presence || {},
+        },
+      }
+    end
+
+  private
```

Comment:
Maybe when it's time to refactor (soon!) we could push some of these methods up into the `Decorator` and share that? Feels a little bit more testable too.

Reply:
Yes, many of those are standalone and can be tested in isolation in a shared class for both APIs, in cleanup PR to follow this.